wkcuber/image_readers.py

```diff
@@ -302,115 +415,209 @@ def find_count_of_axis(czi_file: CziFile, axis: str) -> int:
-                if channel_index is not None:
-                    data = data[:, :, channel_index : channel_index + 1]
-                return data
+            z_file_start = czi_file.start[0] if self.existing_axes[0] == 1 else 0
+            c_axis_index = self.get_access_index_from_existence_array(1)
+            c_file_start = (
+                czi_file.start[c_axis_index] if self.existing_axes[1] == 1 else 0
+            )
+            zero_axis_index = self.get_access_index_from_existence_array(4)
+            zero_file_start = czi_file.start[zero_axis_index]
+            output_channel_offset = 0
+            # Dict[int, Dict[Tuple[int, int], Dict[int, Tuple[int, int]]]]
+            for c_index, c_dict in self.select_correct_tiles(
+                selected_channel, self.tile_shape, czi_file.shape
+            ).items():
+                for c_slice, zero_dict in c_dict.items():
+                    for zero_index, zero_slice in zero_dict.items():
+                        for (
+                            entry
+                        ) in (
+                            czi_file.filtered_subblock_directory  # pylint: disable=not-an-iterable
+                        ):
+                            if (
+                                self.matches_if_exist(
+                                    0, entry.start[0] - z_file_start, z_slice
+                                )
+                                and self.matches_if_exist(
+                                    1,
+                                    (entry.start[c_axis_index] - c_file_start)
+                                    // self.tile_shape[c_axis_index],
+                                    c_index,
+                                )
+                                and self.matches_if_exist(
+                                    4,
+                                    (entry.start[zero_axis_index] - zero_file_start)
+                                    // self.tile_shape[zero_axis_index],
+                                    zero_index,
+                                )
+                            ):
+                                data = entry.data_segment().data()
+                                data = to_target_datatype(data, dtype)
+                                for i in range(c_slice[0], c_slice[1]):
+                                    # format of curr data can be (Z=1),Y,X,0, but should be 0,X,Y
+                                    curr_data = (
+                                        np.take(data, i, c_axis_index)
+                                        if self.existing_axes[1] == 1
+                                        else data
+                                    )
+                                    if curr_data.ndim == 5:
+                                        curr_data = curr_data.transpose(
+                                            (0, 1, -1, -2, -3)
+                                        )
```

Comment:
The comment above suggests that `curr_data`can be (up to?) 4 dimensions, but here are 5 dimensions. Is the `C` missing in the comment above?